src/test/java/io/appium/java_client/android/KeyCodeTest.java

```diff
@@ -16,28 +16,71 @@
 
 package io.appium.java_client.android;
 
+import static org.junit.Assert.fail;
+
+import io.appium.java_client.android.nativekey.AndroidKey;
+import io.appium.java_client.android.nativekey.KeyEvent;
+import io.appium.java_client.android.nativekey.KeyEventFlag;
+import io.appium.java_client.android.nativekey.KeyEventMetaModifier;
 import org.junit.Before;
 import org.junit.Test;
+import org.openqa.selenium.WebDriverException;
 
 public class KeyCodeTest extends BaseAndroidTest {
 
-    @Before public void setup() {
+    @Before
+    public void setUp() {
         driver.resetApp();
     }
 
-    @Test public void pressKeyCodeTest() {
-        driver.pressKeyCode(AndroidKeyCode.HOME);
+    @Test
+    public void pressKeyCodeTest() {
+        try {
+            driver.pressKey(new KeyEvent(AndroidKey.HOME));
+        } catch (WebDriverException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void pressKeyCodeWithMetastateTest() {
+        try {
+            driver.pressKey(new KeyEvent(AndroidKey.SPACE)
+                    .withMetaModifier(KeyEventMetaModifier.SHIFT_ON));
+        } catch (WebDriverException e) {
+            fail(e.getMessage());
```

Comment:
You can check the result of command on ```io.appium.android.apis/.text.KeyEventText``` activity

Reply:
done